include/targets/cuda/shared_memory_cache_helper.h

```diff
@@ -256,6 +285,67 @@ namespace quda
        @brief Synchronize the cache
     */
     __device__ __host__ void sync() { target::dispatch<sync_impl>(); }
+
+    /**
+       @brief Cast operator to allow cache objects to be used where T
+       is expected
+     */
+    __device__ __host__ operator T() const { return load(); }
+
+    /**
+       @brief Assignment operator to allow cache objects to be used on
+       the lhs where T is otherwise expected.
+     */
+    __device__ __host__ void operator=(const T& src) { save(src); }
```

Comment:
Can this also have `const` applied to it?

Reply:
4334841b9ced537246a6d2a4565ae283ac291055